core/src/main/scala/ai/h2o/sparkling/frame/H2OFrame.scala

```diff
@@ -53,11 +53,15 @@ class H2OFrame private(val frameId: String, val columns: Array[H2OColumn], val c
     val selectedIndices = columns.map { name =>
       indices.getOrElse(name, throw new IllegalArgumentException(s"Column $name does not exist in the frame $frameId"))
     }
-    val params = Map(
-      "ast" -> MessageFormat.format(s"( assign {0} (:= {0} (as.factor (cols {0} {1})) {1} []))", frameId, util.Arrays.toString(selectedIndices))
-    )
-    val rapidsFrameV3 = update[RapidsFrameV3](endpoint, "99/Rapids", conf, params)
-    H2OFrame(rapidsFrameV3.key.name)
+    if (selectedIndices.isEmpty) {
```

Comment:
No need to trigger rest request if there are no columns to convert to categoricals